### PR TITLE
feat(menu): checkbox

### PR DESCRIPTION
**Pull request by BerkieBb**
It works, but the style is not good, so I'll let frontend-god Luke finish this

### CODE REVIEW

File: web/src/features/menu/list/index.tsx [hunk 1]
```diff
@@ -59,41 +61,71 @@ const ListMenu: React.FC = () => {
...
+              indexStates[selected] - 1 >= 0
+                ? indexStates[selected] - 1
+                : menu.items[selected].values?.length! - 1
+          });
+        } else if (menu.items[selected].checked !== undefined) {
+          setCheckedStates({
+            ...checkedStates,
+            [selected]: !checkedStates[selected]
+          });
+        }
```
Comment (LukeWasTakenn) [inline web/src/features/menu/list/index.tsx:93]: I don't know how I feel about left-right arrows checking/unchecking the checkbox, imo this should be handled by `Enter`, then just make it not close the menu in the process.

While left-right arrows to check/uncheck are cool it adds unnecessarily complexity to the code and most people's first instinct would be to press enter to toggle the checkbox.

Comment (BerkieBb) [reply web/src/features/menu/list/index.tsx:93]: Well, both enter and left/right handle it

Comment (LukeWasTakenn) [reply web/src/features/menu/list/index.tsx:93]: yes but not returning out of the confirm function doesn't give the user that feedback that the checkbox changed

Comment (BerkieBb) [reply web/src/features/menu/list/index.tsx:93]: We've got the onCheck function for that

**Pull request merged**